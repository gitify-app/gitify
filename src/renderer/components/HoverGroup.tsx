import type { FC, ReactNode } from 'react';

interface IHoverGroup {
  children: ReactNode;
}

export const HoverGroup: FC<IHoverGroup> = ({ children }: IHoverGroup) => {
  return (
    <div className="flex items-center justify-center gap-2 opacity-0 transition-opacity group-hover:opacity-80">
      {children}
    </div>
  );
};
