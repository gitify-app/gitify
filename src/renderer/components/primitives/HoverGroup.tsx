import type { FC, ReactNode } from 'react';

import { Stack } from '@primer/react';
import { cn } from '../../utils/cn';

interface IHoverGroup {
  children: ReactNode;
  bgColor:
    | 'group-hover:bg-gitify-account-rest'
    | 'group-hover:bg-gitify-repository'
    | 'group-hover:bg-gitify-notification-hover';
}

export const HoverGroup: FC<IHoverGroup> = ({
  bgColor,
  children,
}: IHoverGroup) => {
  return (
    <Stack
      direction="horizontal"
      align="center"
      gap="none"
      className={cn(
        'absolute right-0 h-full',
        'opacity-0 transition-opacity group-hover:opacity-100',
        bgColor,
      )}
    >
      {children}
    </Stack>
  );
};
