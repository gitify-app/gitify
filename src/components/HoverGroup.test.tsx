import { render } from '@testing-library/react';
import { HoverGroup } from './HoverGroup';

describe('components/HoverGroup.tsx', () => {
  it('should render', () => {
    const tree = render(<HoverGroup>Hover Group</HoverGroup>);
    expect(tree).toMatchSnapshot();
  });
});
