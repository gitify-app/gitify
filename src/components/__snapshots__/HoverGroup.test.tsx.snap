// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/HoverGroup.tsx should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center justify-center gap-2 opacity-0 transition-opacity group-hover:opacity-80"
      >
        Hover Group
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-center gap-2 opacity-0 transition-opacity group-hover:opacity-80"
    >
      Hover Group
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
